### PR TITLE
Remove deprecated MOCK_METHODx calls

### PR DESCRIPTION
**Pull request by dkwingsmt**
The issue that blocks these mocking methods have been resolved.

However, there is [one more mention of this issue in `run_tests.py`](https://github.com/flutter/engine/blob/8f1b77b1ac693524435fe9199ce2e0cb43662bd3/testing/run_tests.py#L400) and I have no idea why it was marked so. It was originally added [here](https://github.com/flutter/engine/pull/19033/files#diff-521d7c59c3f721b259f094c760c2613e16e889b0f24702280924466109e3b0b0R137).

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by chinmaygarde — APPROVED** [review]
There are tons of these everywhere in the engine still. Should we just fix all of those too?

**Comment by dkwingsmt** [comment]
Sure! Let me do a full search.

File: impeller/renderer/testing/mocks.h [hunk 1]
```diff
@@ -17,111 +17,156 @@ namespace testing {
 class MockDeviceBuffer : public DeviceBuffer {
  public:
   MockDeviceBuffer(const DeviceBufferDescriptor& desc) : DeviceBuffer(desc) {}
-  MOCK_METHOD3(CopyHostBuffer,
-               bool(const uint8_t* source, Range source_range, size_t offset));
+  // MOCK_METHOD(bool, CopyHostBuffer, (const uint8_t* source, Range
+  // source_range, size_t offset), (override));
```
Comment (dkwingsmt) [inline impeller/renderer/testing/mocks.h:21]: This method is not virtual. Should we still mock it?

File: impeller/renderer/testing/mocks.h [hunk 2]
```diff
@@ -17,111 +17,156 @@ namespace testing {
...
-  MOCK_CONST_METHOD0(IsValid, bool());
-  MOCK_CONST_METHOD0(GetSize, ISize());
-  MOCK_METHOD3(OnSetContents,
-               bool(const uint8_t* contents, size_t length, size_t slice));
-  MOCK_METHOD2(OnSetContents,
-               bool(std::shared_ptr<const fml::Mapping> mapping, size_t slice));
+  MOCK_METHOD(void, SetLabel, (std::string_view label), (override));
+  // MOCK_METHOD(bool, SetContents, (const uint8_t* contents, size_t length,
+  // size_t slice), (override)); MOCK_METHOD(bool, SetContents,
+  // (std::shared_ptr<const fml::Mapping> mapping, size_t slice), (override));
```
Comment (dkwingsmt) [inline impeller/renderer/testing/mocks.h:159]: This method is not virtual. Should we still mock it?

**Comment by dkwingsmt** [comment]
@chinmaygarde I've converted all current cases using regex. Can you take the two cases that I marked with comment? They're not virtual. I think we don't have to mock them. If we do, then the cookbook requires we use [a different approach](http://google.github.io/googletest/gmock_cook_book.html).

File: shell/platform/windows/flutter_windows_view.h [hunk 3]
```diff
@@ -59,7 +59,7 @@ class FlutterWindowsView : public WindowBindingHandlerDelegate,
   WindowsRenderTarget* GetRenderTarget() const;
 
   // Return the currently configured PlatformWindow.
-  PlatformWindow GetPlatformWindow() const;
+  virtual PlatformWindow GetPlatformWindow() const;
```
Comment (dkwingsmt) [inline shell/platform/windows/flutter_windows_view.h:62]: Also this method is now marked as virtual, since its mock is used.

**Review by chinmaygarde — APPROVED** [review]
Looks like the presubs are mad at a couple of Windows unittests. They seem straightforward though.

Thanks so much for doing this!

**Pull request merged**